internal/spoa.go

```diff
@@ -128,84 +154,34 @@ func (s *SPOA) readHeaders(headers string) (http.Header, error) {
 	return h, nil
 }
 
-func (s *SPOA) cleanApplications() {
-	for _, app := range s.applications {
-		if err := app.logger.Sync(); err != nil {
-			app.logger.Error("failed to sync logger", zap.Error(err))
-		}
-	}
-}
-
-func logError(logger *zap.Logger) ErrorLogCallback {
-	return func(mr types.MatchedRule) {
-		data := mr.ErrorLog()
-		switch mr.Rule().Severity() {
-		case types.RuleSeverityEmergency:
-			logger.Error(data)
-		case types.RuleSeverityAlert:
-			logger.Error(data)
-		case types.RuleSeverityCritical:
-			logger.Error(data)
-		case types.RuleSeverityError:
-			logger.Error(data)
-		case types.RuleSeverityWarning:
-			logger.Warn(data)
-		case types.RuleSeverityNotice:
-			logger.Info(data)
-		case types.RuleSeverityInfo:
-			logger.Info(data)
-		case types.RuleSeverityDebug:
-			logger.Debug(data)
-		}
-	}
-}
-
 // New Create a new SPOA instance.
 func New(conf *config.Config) (*SPOA, error) {
 	apps := make(map[string]*application)
 	for name, cfg := range conf.Applications {
-		pe := zap.NewProductionEncoderConfig()
-
-		fileEncoder := zapcore.NewJSONEncoder(pe)
-
-		pe.EncodeTime = zapcore.ISO8601TimeEncoder
+		wafConf := coraza.NewWAFConfig().
```

Comment:
I'd prefer we get the platform logging and plug into an implementation coraza can use. For example we use the zap logger created by caddy and plug into coraza https://github.com/corazawaf/coraza-caddy/blob/main/coraza.go#L54

Reply:
TBH, I had tried it before pushed the PR. It doesn't suit well.